components/gantry/oneaxis/oneaxis.go

```diff
@@ -37,45 +37,29 @@ type Config struct {
 
 // Validate ensures all parts of the config are valid.
 func (cfg *Config) Validate(path string) ([]string, error) {
-	deps, err := cfg.validate()
-	if err != nil {
-		err = utils.NewConfigValidationError(path, err)
-	}
-	return deps, err
-}
-
-func (cfg *Config) validate() ([]string, error) {
 	var deps []string
 
 	if len(cfg.Motor) == 0 {
-		return nil, errors.New("cannot find motor for gantry")
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "motor")
 	}
 	deps = append(deps, cfg.Motor)
 
 	if cfg.LengthMm <= 0 {
-		return nil, errors.New("each axis needs a non-zero and positive length")
-	}
-
-	if len(cfg.LimitSwitchPins) == 0 && len(cfg.Board) > 0 {
```

Comment:
[q] why is it no longer necessary to check this combination of `LimitSwitchPins` and `Board`?